profiler/profiler.go

```diff
@@ -67,18 +65,24 @@ func Stop() {
 // profiler collects and sends preset profiles to the Datadog API at a given frequency
 // using a given configuration.
 type profiler struct {
-	cfg        *config           // profile configuration
-	out        chan batch        // upload queue
-	uploadFunc func(batch) error // defaults to (*profiler).upload; replaced in tests
-	exit       chan struct{}     // exit signals the profiler to stop; it is closed after stopping
+	cfg         *config // profile configuration
+	targetURL   string  // url to which we'll try uploading
```

Comment:
This is config, not profiler internals. Can we put it in `cfg`? The `Start` function could just set the `targetURL` correctly and then we can also remove the `targetURL()` method.